File: cpp/ecbenchmark/src/cec10/CecFunction.cpp
#include "ecbenchmark/cec10/CecFunction.h"

namespace ecb {
    namespace cec10 {

        CecFunction::CecFunction(const std::string& name, const std::string& description,
                int dimensions, scalar minimum, scalar maximum, bool minimization,
                int mValue, CecRandom* randomizer)
        : AbstractFunction(name, description, dimensions, minimum, maximum, minimization),
        _mValue(mValue), _randomizer(randomizer) {

        }

        CecFunction::~CecFunction() {
            delete _randomizer;
        }

        void CecFunction::setMValue(int m) {
            this->_mValue = m;
        }

        int CecFunction::mValue() const {
            return this->_mValue;
        }

        void CecFunction::setRandomizer(CecRandom* randomizer) {
            this->_randomizer = randomizer;
        }

        CecRandom* CecFunction::randomizer() const {
            return this->_randomizer;
        }
    }

}

File: java/ecbenchmark/src/ecbenchmark/cec10/CecFunction.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.AbstractFunction;
import java.util.Random;

/**
 *
 * @author jcrada
 */
public abstract class CecFunction extends AbstractFunction {

    protected Random randomizer = null;
    protected int mValue = 50;

    public CecFunction(String name, String description, int dimensions,
            double minimum, double maximum, boolean minimization, Random randomizer) {
        super(name, description, dimensions, minimum, maximum, minimization);
        this.randomizer = randomizer;
    }

    public CecFunction(String name, String description, int dimensions,
            double minimum, double maximum, boolean minimization, Random randomizer,
            int mValue) {
        this(name, description, dimensions, minimum, maximum, minimization, randomizer);
        this.mValue = mValue;
    }

    public void setRandomizer(Random random) {
        this.randomizer = random;
    }

    public Random getRandomizer() {
        return this.randomizer;
    }

    public void setMValue(int m) {
        this.mValue = m;
    }

    public int getMValue() {
        return this.mValue;
    }
}
